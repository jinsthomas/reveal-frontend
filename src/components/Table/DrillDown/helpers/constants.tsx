/** Constants */
import React from 'react';

export const ID = 'id';
export type ID = typeof ID;

export const PARENT_ID = 'parent_id';
export type PARENT_ID = typeof PARENT_ID;

export const ROOT_PARENT_ID = null;
export type ROOT_PARENT_ID = typeof ROOT_PARENT_ID;

export const LINKER_ITEM_CSS_CLASS = 'dd-linker-item';
export type LINKER_ITEM_CSS_CLASS = typeof LINKER_ITEM_CSS_CLASS;

export const CLICKABLE_CSS_CLASS = 'dd-linker-item dd-clickable';
export type CLICKABLE_CSS_CLASS = typeof CLICKABLE_CSS_CLASS;

export const CARET_CSS_CLASS = 'dd-caret';
export type CARET_CSS_CLASS = typeof CARET_CSS_CLASS;

export const CARET_SPAN = <span className={CARET_CSS_CLASS}>&nbsp;&#9660;</span>;
export type CARET_SPAN = typeof CARET_SPAN;

export const DEFAULT_ROW_HEIGHT = '2em';
export type DEFAULT_ROW_HEIGHT = typeof DEFAULT_ROW_HEIGHT;
