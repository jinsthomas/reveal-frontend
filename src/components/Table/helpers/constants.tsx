/** Constants */
import React from 'react';
export const ID = 'id';
export type ID = typeof ID;

export const PARENT_ID = 'parent_id';
export type PARENT_ID = typeof PARENT_ID;

export const ROOT_PARENT_ID = null;
export type ROOT_PARENT_ID = typeof ROOT_PARENT_ID;

export const LINKER_ITEM_CSS_CLASS = 'dd-linker-item';
export type LINKER_ITEM_CSS_CLASS = typeof LINKER_ITEM_CSS_CLASS;

export const CLICKABLE_CSS_CLASS = 'dd-linker-item dd-clickable';
export type CLICKABLE_CSS_CLASS = typeof CLICKABLE_CSS_CLASS;

export const CARET_CSS_CLASS = 'dd-caret';
export type CARET_CSS_CLASS = typeof CARET_CSS_CLASS;

export const CARET = <span className={CARET_CSS_CLASS}>&nbsp;&#9660;</span>;
export type CARET = typeof CARET;

export const ROWS_TO_DISPLAY = 'Rows to display';
export type ROWS_TO_DISPLAY = typeof ROWS_TO_DISPLAY;

export const PAGE = 'page';
export type PAGE = typeof PAGE;

export const OF = 'of';
export type OF = typeof OF;
