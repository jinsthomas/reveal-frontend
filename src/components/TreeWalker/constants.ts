export const ACTIVE = 'Active';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
