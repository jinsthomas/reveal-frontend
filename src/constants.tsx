// env vars
export const WEBSITE_NAME = process.env.REACT_APP_WEBSITE_NAME;
export type WEBSITE_NAME = typeof WEBSITE_NAME;

// strings
export const LOCATION = 'Location';
export type LOCATION = typeof LOCATION;
export const PROVINCE = 'Province';
export type PROVINCE = typeof PROVINCE;
export const HOME = 'Home';
export type HOME = typeof HOME;
export const FOCUS_INVESTIGATION = 'Focus Investigation';
export type FOCUS_INVESTIGATION = typeof FOCUS_INVESTIGATION;
export const FOCUS_INVESTIGATIONS = 'Focus Investigations';
export type FOCUS_INVESTIGATIONS = typeof FOCUS_INVESTIGATIONS;

// internal urls
export const HOME_URL = '/';
export type HOME_URL = typeof HOME_URL;
export const IRS_URL = '/irs';
export type IRS_URL = typeof IRS_URL;
export const FI_URL = '/focus-investigation';
export type FI_URL = typeof FI_URL;
export const FI_HISTORICAL_URL = '/focus-investigation/historical';
export type FI_HISTORICAL_URL = typeof FI_HISTORICAL_URL;

// colors
export const GREEN = 'Green';
export type GREEN = typeof GREEN;
export const YELLOW = 'Yellow';
export type YELLOW = typeof YELLOW;
export const ORANGE = 'Orange';
export type ORANGE = typeof ORANGE;
export const RED = 'Red';
export type RED = typeof RED;

// thresholds
export const GREEN_THRESHOLD = 0.9;
export type GREEN_THRESHOLD = typeof GREEN_THRESHOLD;
export const YELLOW_THRESHOLD = 0.2;
export type YELLOW_THRESHOLD = typeof YELLOW_THRESHOLD;
export const ORANGE_THRESHOLD = 0.8;
export type ORANGE_THRESHOLD = typeof ORANGE_THRESHOLD;

// 1-3-7 thresholds
export const THRESHOLD_137 = 0;
