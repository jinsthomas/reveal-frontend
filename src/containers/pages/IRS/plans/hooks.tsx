import { SupersetFormData } from '@onaio/superset-connector/dist/types';
import React, { useEffect, useState } from 'react';
import { ActionCreator } from 'redux';
import { SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { displayError } from '../../../../helpers/errors';
import supersetFetch from '../../../../services/superset';

interface AsyncSupersetPlans<TData, TAction> {
  superset: typeof supersetFetch;
  data: TData[];
  fetchPlans: ActionCreator<TAction>;
  supersetSlice: string;
  supersetOptions: SupersetFormData | null;
}

export const useAsyncSupersetPlans = <TData, TAction>({
  superset = supersetFetch,
  data = [],
  fetchPlans,
  supersetSlice = SUPERSET_IRS_REPORTING_PLANS_SLICE,
  supersetOptions = null,
}: AsyncSupersetPlans<TData, TAction>) => {
  const [loading, setLoading] = useState<boolean>(false);
  const mounted = React.useRef<boolean>(false);

  /** async function to load the data */
  async function loadData() {
    try {
      setLoading(data.length === 0); // only set loading when there are no plans
      const asyncOperation = supersetOptions
        ? superset(supersetSlice, supersetOptions)
        : superset(supersetSlice);

      await asyncOperation.then((result: TData[]) => fetchPlans(result));
    } catch (e) {
      throw e;
    } finally {
      if (mounted) {
        setLoading(false);
      }
    }
  }

  useEffect(() => {
    mounted.current = true;
    loadData().catch(error => displayError(error));
    return () => {
      mounted.current = false;
    };
  }, []);

  return loading;
};
