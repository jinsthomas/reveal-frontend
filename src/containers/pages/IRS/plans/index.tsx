import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import { SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { IRS_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_IRS_PLAN_URL } from '../../../../constants';
import { getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import IRSPlansReducer, {
  fetchIRSPlans,
  GenericPlan,
  makeIRSPlansArraySelector,
  reducerName as IRSPlansReducerName,
} from '../../../../store/ducks/generic/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the IRS plan definitions reducer */
reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);

/** interface for IRS PlanList props */
interface PlanListProps extends GenericPlanListProps {
  ownProps: any;
}

/** Simple component that loads a privew list of IRS plans */
const IRSPlansList = (props: PlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for IRSPlansList */
const defaultProps: PlanListProps = {
  fetchPlans: fetchIRSPlans,
  ownProps: {},
  pageTitle: IRS_PLANS,
  pageUrl: REPORT_IRS_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_IRS_REPORTING_PLANS_SLICE,
};

IRSPlansList.defaultProps = defaultProps;

export { IRSPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const plans = makeIRSPlansArraySelector()(state, { plan_title: searchedTitle });

  return {
    ...ownProps,
    plans,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchIRSPlans };

/** Connected ActiveFI component */
const ConnectedIRSPlansList = connect(mapStateToProps, mapDispatchToProps)(IRSPlansList);

export default ConnectedIRSPlansList;
