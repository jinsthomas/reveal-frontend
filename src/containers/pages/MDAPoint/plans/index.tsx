import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import { SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { MDA_POINT_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_POINT_PLAN_URL } from '../../../../constants';
import { getQueryParams } from '../../../../helpers/utils';
import { RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import MDAPointPlansReducer, {
  fetchMDAPointPlans,
  makeMDAPointPlansArraySelector,
  MDAPointPlan,
  reducerName as MDAPointPlansReducerName,
} from '../../../../store/ducks/generic/MDAPointPlans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(MDAPointPlansReducerName, MDAPointPlansReducer);

// /** interface for MDA PlanList props */
interface PlanListProps extends GenericPlanListProps {
  ownProps: any;
}

/** Simple component that loads a preview list of MDA plans */
const MDAPointPlansList = (props: PlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDAPointPlansList */
const defaultProps: PlanListProps = {
  fetchPlans: fetchMDAPointPlans,
  ownProps: {},
  pageTitle: MDA_POINT_PLANS,
  pageUrl: REPORT_MDA_POINT_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE,
};

MDAPointPlansList.defaultProps = defaultProps;

export { MDAPointPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: MDAPointPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const MDAPointPlansArray = makeMDAPointPlansArraySelector()(state, { plan_title: searchedTitle });

  return {
    ...ownProps,
    plans: MDAPointPlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchMDAPointPlans };

/** Connected ActiveFI component */
const ConnectedMDAPointPlansList = connect(mapStateToProps, mapDispatchToProps)(MDAPointPlansList);

export default ConnectedMDAPointPlansList;
