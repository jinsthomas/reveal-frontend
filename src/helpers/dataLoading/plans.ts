import { toast } from 'react-toastify';
import { ActionCreator, AnyAction } from 'redux';
import { OPENSRP_PLANS_BY_USER_FILTER } from '../../constants';
import { OpenSRPService } from '../../services/opensrp';
import store from '../../store';
import { fetchPlansByUser, FetchPlansByUserAction } from '../../store/ducks/opensrp/planIdsByUser';
import { growl } from '../utils';

/** find plans that the given user has access to
 * @param {string} userName - openMRSUserName
 * @param {ActionCreator<FetchPlansByUserAction>} actionCreator - action creator for adding practitioners to store
 * @param {typeof OpenSRPService} service -  the OpenSRP service
 * @param {ActionCreator<AnyAction>} responseActionCreator - optionalActionCreator to dispatch the planData in response
 */
export async function loadPlansByUserFilter<T>(
  userName: string,
  actionCreator: ActionCreator<FetchPlansByUserAction> = fetchPlansByUser,
  service: typeof OpenSRPService = OpenSRPService,
  responseActionCreator?: ActionCreator<AnyAction>
) {
  const serve = new service(`${OPENSRP_PLANS_BY_USER_FILTER}/${userName}`);
  serve
    .list()
    .then((response: T[]) => {
      store.dispatch(actionCreator(response, userName));
      if (responseActionCreator) {
        store.dispatch(responseActionCreator(response));
      }
    })
    .catch((err: Error) => {
      growl(err.message, { type: toast.TYPE.ERROR });
    });
}
