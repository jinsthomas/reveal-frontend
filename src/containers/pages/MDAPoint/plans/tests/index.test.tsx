import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { REPORT_MDA_POINT_PLAN_URL } from '../../../../../constants';
import store from '../../../../../store';
import { fetchMDAPointPlans } from '../../../../../store/ducks/generic/MDAPointPlans';
import { GenericPlan } from '../../../../../store/ducks/generic/plans';
import * as fixtures from '../../../../../store/ducks/generic/tests/fixtures';
import ConnectedMDAPointPlansList, { MDAPointPlansList } from '../index';

jest.mock('../../../../../configs/env');
/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

const history = createBrowserHistory();

const props = {
  history,
  location: {
    hash: '',
    pathname: REPORT_MDA_POINT_PLAN_URL,
    search: '',
    state: undefined,
  },
  match: {
    isExact: true,
    params: {},
    path: `${REPORT_MDA_POINT_PLAN_URL}/`,
    url: `${REPORT_MDA_POINT_PLAN_URL}/`,
  },
};

describe('components/IRS Reports/IRSPlansList', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    shallow(
      <Router history={history}>
        <MDAPointPlansList {...props} />
      </Router>
    );
  });

  it('renders plan definition list correctly', () => {
    store.dispatch(fetchMDAPointPlans(fixtures.MDAPointplans as GenericPlan[]));
    fetch.mockResponseOnce(JSON.stringify({}));
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedMDAPointPlansList {...props} />
        </Router>
      </Provider>
    );
    expect(toJson(wrapper.find('BreadcrumbItem li'))).toMatchSnapshot('breadcrumbs');
    expect(toJson(wrapper.find('h3.page-title'))).toMatchSnapshot('page title');
    expect(toJson(wrapper.find('thead tr th'))).toMatchSnapshot('table headers');
    expect(toJson(wrapper.find('tbody tr td'))).toMatchSnapshot('table rows');

    expect(wrapper.find('GenericPlansList').length).toBe(1);
    expect(wrapper.find('GenericPlansList').props()).toMatchSnapshot('GenericPlansList props');

    expect(wrapper.find('.page-title').text()).toEqual('MDA Point Plans');

    wrapper.unmount();
  });

  it('handles search correctly', async () => {
    store.dispatch(fetchMDAPointPlans(fixtures.MDAPointplans as GenericPlan[]));
    props.location.search = '?title=Berg';
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedMDAPointPlansList {...props} />
        </Router>
      </Provider>
    );
    expect(wrapper.find('tbody tr td Link').text()).toEqual('Berg Eswatini 2019');

    expect((wrapper.find('GenericPlansList').props() as any).plans).toMatchSnapshot(
      'search results props'
    );
  });
});
