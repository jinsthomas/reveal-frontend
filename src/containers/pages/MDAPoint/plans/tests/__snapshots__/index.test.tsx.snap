// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IRS Reports/IRSPlansList handles search correctly: search results props 1`] = `
Array [
  Object {
    "jurisdiction_root_parent_ids": Array [
      "2942",
    ],
    "plan_date": "2019-08-29",
    "plan_effective_period_end": "2019-09-18",
    "plan_effective_period_start": "2019-08-29",
    "plan_id": "c6e59ec1-8c54-5fdf-acf3-ee3692983f38",
    "plan_intervention_type": "MDA-Point",
    "plan_name": "Berg-Eswatini-2019",
    "plan_status": "active",
    "plan_title": "Berg Eswatini 2019",
    "plan_version": "2",
  },
]
`;

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: GenericPlansList props 1`] = `
Object {
  "fetchPlans": [Function],
  "history": Object {
    "action": "POP",
    "block": [Function],
    "createHref": [Function],
    "go": [Function],
    "goBack": [Function],
    "goForward": [Function],
    "length": 1,
    "listen": [Function],
    "location": Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    },
    "push": [Function],
    "replace": [Function],
  },
  "location": Object {
    "hash": "",
    "pathname": "/intervention/mda-point/report",
    "search": "",
    "state": undefined,
  },
  "match": Object {
    "isExact": true,
    "params": Object {},
    "path": "/intervention/mda-point/report/",
    "url": "/intervention/mda-point/report/",
  },
  "pageTitle": "MDA Point Plans",
  "pageUrl": "/intervention/mda-point/report",
  "plans": Array [
    Object {
      "jurisdiction_root_parent_ids": Array [
        "2942",
      ],
      "plan_date": "2019-09-05",
      "plan_effective_period_end": "2019-09-25",
      "plan_effective_period_start": "2019-09-05",
      "plan_id": "40357eff-81b6-4e32-bd3d-484019689f7c",
      "plan_intervention_type": "MDA-Point",
      "plan_name": "MDA-Point-2019-09-05-TEST",
      "plan_status": "retired",
      "plan_title": "MDA-Point-2019-09-05 TEST",
      "plan_version": "2",
    },
    Object {
      "jurisdiction_root_parent_ids": Array [
        "2942",
      ],
      "plan_date": "2019-08-30",
      "plan_effective_period_end": "2019-09-14",
      "plan_effective_period_start": "2019-08-30",
      "plan_id": "9f19b77c-b9a5-5832-a4e5-4b461d18fce7",
      "plan_intervention_type": "MDA-Point",
      "plan_name": "MDA-Point-2019-08-30m",
      "plan_status": "active",
      "plan_title": "MDA-Point-2019-08-30m",
      "plan_version": "2",
    },
    Object {
      "jurisdiction_root_parent_ids": Array [
        "2942",
      ],
      "plan_date": "2019-08-29",
      "plan_effective_period_end": "2019-09-18",
      "plan_effective_period_start": "2019-08-29",
      "plan_id": "c6e59ec1-8c54-5fdf-acf3-ee3692983f38",
      "plan_intervention_type": "MDA-Point",
      "plan_name": "Berg-Eswatini-2019",
      "plan_status": "active",
      "plan_title": "Berg Eswatini 2019",
      "plan_version": "2",
    },
  ],
  "service": [Function],
  "supersetReportingSlice": "01",
}
`;

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: breadcrumbs 1`] = `
Array [
  <li
    className="breadcrumb-item"
  >
    <Link
      key="0"
      replace={false}
      to="/"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </Link>
  </li>,
  <li
    aria-current="page"
    className="active breadcrumb-item"
  >
    MDA Point Plans
  </li>,
]
`;

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: page title 1`] = `
<h3
  className="mt-3 mb-3 page-title"
>
  MDA Point Plans
</h3>
`;

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: table headers 1`] = `
Array [
  <th
    className="listview-td"
    key="0"
  >
    Title
  </th>,
  <th
    className="listview-td"
    key="1"
  >
    Date Created
  </th>,
  <th
    className="listview-td"
    key="2"
  >
    Start Date
  </th>,
  <th
    className="listview-td"
    key="3"
  >
    End Date
  </th>,
  <th
    className="listview-td"
    key="4"
  >
    Status
  </th>,
]
`;

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: table rows 1`] = `
Array [
  <td
    className="listview-th"
    key="0"
  >
    <Link
      key="40357eff-81b6-4e32-bd3d-484019689f7c"
      replace={false}
      to="/intervention/mda-point/report/40357eff-81b6-4e32-bd3d-484019689f7c"
    >
      <a
        href="/intervention/mda-point/report/40357eff-81b6-4e32-bd3d-484019689f7c"
        onClick={[Function]}
      >
        MDA-Point-2019-09-05 TEST
      </a>
    </Link>
  </td>,
  <td
    className="listview-th"
    key="1"
  >
    2019-09-05
  </td>,
  <td
    className="listview-th"
    key="2"
  >
    2019-09-05
  </td>,
  <td
    className="listview-th"
    key="3"
  >
    2019-09-25
  </td>,
  <td
    className="listview-th"
    key="4"
  >
    retired
  </td>,
  <td
    className="listview-th"
    key="0"
  >
    <Link
      key="9f19b77c-b9a5-5832-a4e5-4b461d18fce7"
      replace={false}
      to="/intervention/mda-point/report/9f19b77c-b9a5-5832-a4e5-4b461d18fce7"
    >
      <a
        href="/intervention/mda-point/report/9f19b77c-b9a5-5832-a4e5-4b461d18fce7"
        onClick={[Function]}
      >
        MDA-Point-2019-08-30m
      </a>
    </Link>
  </td>,
  <td
    className="listview-th"
    key="1"
  >
    2019-08-30
  </td>,
  <td
    className="listview-th"
    key="2"
  >
    2019-08-30
  </td>,
  <td
    className="listview-th"
    key="3"
  >
    2019-09-14
  </td>,
  <td
    className="listview-th"
    key="4"
  >
    active
  </td>,
  <td
    className="listview-th"
    key="0"
  >
    <Link
      key="c6e59ec1-8c54-5fdf-acf3-ee3692983f38"
      replace={false}
      to="/intervention/mda-point/report/c6e59ec1-8c54-5fdf-acf3-ee3692983f38"
    >
      <a
        href="/intervention/mda-point/report/c6e59ec1-8c54-5fdf-acf3-ee3692983f38"
        onClick={[Function]}
      >
        Berg Eswatini 2019
      </a>
    </Link>
  </td>,
  <td
    className="listview-th"
    key="1"
  >
    2019-08-29
  </td>,
  <td
    className="listview-th"
    key="2"
  >
    2019-08-29
  </td>,
  <td
    className="listview-th"
    key="3"
  >
    2019-09-18
  </td>,
  <td
    className="listview-th"
    key="4"
  >
    active
  </td>,
]
`;
